nix/haskell.nix

```diff
@@ -52,8 +54,6 @@ haskell-nix.cabalProject' ({ pkgs
       hlint
```

Comment:
out of the diff above, but I suggest we also take this opportunity to remove all the `packages` arguments to `shellFor`, they shouldn't be needed

Reply:
Sorted